extensions/src/lock/CrossPlatformLock.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { open, close, write, unlink } from "fs";
+import { promisify } from "util";
+import { pid } from "process";
+import { CrossPlatformLockOptions } from "./CrossPlatformLockOptions";
+import { Constants } from "../utils/Constants";
+import { PersistenceError } from "../error/PersistenceError";
+
+/**
+ * Cross-process lock that works on all platforms.
+ */
+export class CrossPlatformLock {
+
+    private readonly lockFilePath: string;
+    private lockFileDescriptor: number;
+    private readonly retryNumber: number;
+    private readonly retryDelay: number;
+
+    constructor(lockFilePath:string, lockOptions?: CrossPlatformLockOptions) {
+        this.lockFilePath = lockFilePath;
+        this.retryNumber = lockOptions ? lockOptions.retryNumber : 500;
+        this.retryDelay = lockOptions ? lockOptions.retryDelay : 100;
+    }
+
+    /**
+     * Locks cache from read or writes by creating file with same path and name as
+     * cache file but with .lockfile extension. If another process has already created
+     * the lockfile, will retry again based on configuration settings set by CrossPlatformLockOptions
+     */
+    public async lock(): Promise<void> {
+        const processId = pid.toString();
+        for (let tryCount = 0; tryCount < this.retryNumber; tryCount++)
+            try {
+                console.log("Pid " + pid + " trying to acquire lock");
```

Comment:
Every console.log statement will be switched over to using the msal-common logger in upcoming PR